cpu/native/mci/native-mci.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) Lucas Jenß
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup native
+ * @{
+ */
+
+/**
+ * @file
+ * @brief   Implements the MCI storage driver by using the flash_sim component
+ *          as backend. It is 8 MiB by default and its size can be increased
+ *          using `NATIVE_MCI_SIZE_MIB` which specifies the MCI storage size in MiB.
+ *
+ *          Additional properties are:
+ *
+ *              Sector (page) size: 512 B
+ *              Erase block size: 512 KiB (524288 Bytes)
+ *
+ * @author  Lucas Jenß <lucas@x3ro.de>
+ *
+ */
+
+#include <string.h>
+#include <stdbool.h>
+#include "diskio.h"
+
+#include "storage/flash_sim.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#ifndef NATIVE_MCI_SIZE_MIB
+#define NATIVE_MCI_SIZE_MIB 8
+#endif
+
+#define NATIVE_MCI_SIZE (1024 * 1024 * NATIVE_MCI_SIZE_MIB)
+#define NATIVE_MCI_SECTOR_SIZE (512)
+#define NATIVE_MCI_BLOCK_SIZE (512 * 1024)
+#define NATIVE_MCI_SECTOR_COUNT (NATIVE_MCI_SIZE / NATIVE_MCI_SECTOR_SIZE)
+
+flash_sim fs;
+
+bool _validate_parameters(unsigned long sector, unsigned char count) {
+    if(count < 1) {
+        return false;
+    }
+
+    if(sector + count > NATIVE_MCI_SECTOR_COUNT) {
+        return false;
+    }
+
+    return true;
+}
+
+diskio_sta_t mci_initialize(void) {
+    fs.page_size = NATIVE_MCI_SECTOR_SIZE;
+    fs.block_size = NATIVE_MCI_BLOCK_SIZE;
+    fs.storage_size = NATIVE_MCI_SIZE;
+
+    if(flash_sim_init(&fs) == 0) {
+        return DISKIO_RES_OK;
+    } else {
+        return DISKIO_RES_ERROR;
+    }
+}
+
+diskio_sta_t mci_status(void) {
+    return DISKIO_RES_OK;
```

Comment:
wrong return type, `DISKIO_RES_OK` is of type `diskio_result_t`, simply return 0 or `0x0` here should do